core/src/main/scala/org/broadinstitute/dsde/rawls/entities/compact/CompactEntityProvider.scala

```diff
@@ -116,9 +119,53 @@ class CompactEntityProvider(requestArguments: EntityRequestArguments, repository
 
   override def deleteEntities(entityRefs: Seq[AttributeEntityReference],
                               parentContext: RawlsRequestContext
-  ): Future[Int] = ???
+  ): Future[Int] =
+    repository.dataSource.inTransaction { _ =>
+      for {
+        entities: Seq[Option[CompactEntityRecord]] <- DBIO.sequence(
+          entityRefs.map(ref => repository.queries.getEntity(workspaceId, ref.entityType, ref.entityName))
+        )
+        entityIds = entities.flatten.map(_.id).toSet
+        // check if any of these entities are referenced by someone else
+        referencingEntities: Seq[AttributeEntityReference] <- repository.queries.getReferencingEntities(entityIds)
+        _ = if (referencingEntities.toSet.diff(entityRefs.toSet).size != 0) {
+          throw new DeleteEntitiesConflictException(referencingEntities.toSet)
+        }
+        // remove all references from these entities
+        _ <- repository.queries.deleteAllReferences(entityIds)
+        res <- repository.queries.batchHide(workspaceId, entityRefs)
+      } yield res.sum
+    }
 
-  override def deleteEntitiesOfType(entityType: String, parentContext: RawlsRequestContext): Future[Int] = ???
+  override def deleteEntitiesOfType(entityType: String, parentContext: RawlsRequestContext): Future[Int] =
+    repository.dataSource.inTransaction { _ =>
+      repository.queries.getEntitiesOfType(workspaceId, entityType).flatMap { entities =>
```

Comment:
Same comment here. Pulling back ALL entities of a given type from the db to Scala just to see if they are deletable feels like a scaling bottleneck. Can you do this all in the database?

Reply:
I would still like to see this change - we shouldn't have to retrieve all entities of this type back to the Scala layer to perform the deletions.